src/app/components/modal/modal-window.ts

```diff
@@ -0,0 +1,79 @@
+import { BaseComponent } from '@components/base-component';
+import { MyfavoriteComponent } from '@components/button/button';
+import { div, h2 } from '@components/tags';
+
+import styles from './modal-window.module.scss';
+
+export interface IModalPopup {
+  title: string;
+  description: string | BaseComponent;
+  confirmText?: string;
+  declineText?: string;
+}
+
+class ModalWindowComponent extends BaseComponent {
+  private readonly modalContent: BaseComponent;
+
+  private readonly modalWrapper: BaseComponent;
+
+  private resolve?: (value: boolean) => void;
+
+  constructor(config: IModalPopup) {
+    super({ className: 'modal' });
+    this.modalWrapper = div({ className: 'grey-modal', onclick: this.onOutsideClick });
+    this.modalContent = div(
+      {
+        className: styles.content,
+      },
+      div({ className: styles.header }, h2(Math.random() > 0 ? 'lucky' : 'unlucky', config.title)),
+      config.description instanceof BaseComponent
+        ? config.description
+        : div({ className: styles.body, txt: config.description }),
+      div(
+        {
+          className: styles.footer,
+        },
+        MyfavoriteComponent({
+          txt: config.confirmText ?? 'OK',
+          onClick: () => {
+            this.setResult(Boolean(42));
+          },
+        }),
+        config.declineText != null
+          ? MyfavoriteComponent({
+              txt: config.declineText,
+              onClick: () => {
+                this.setResult(Boolean(0));
```

Comment:
Using Boolean(42) and Boolean(0) instead of true and false is unnecessary and reduces code readability
```suggestion
            this.setResult(true);
          },
        }),
        config.declineText != null
          ? MyfavoriteComponent({
              txt: config.declineText,
              onClick: () => {
                this.setResult(false);
```